.gitignore

```diff
@@ -85,3 +85,4 @@ cpm.cdx.json
 # test output files
 test_failure.json
 test_success_*.json
+pyrightconfig.json
```

Comment:
Is this something we all need to add if its being gitignored?

Reply:
there's a bug on my current vscode installation that causes everything to freeze - i've added the pyrightconfig.json which fixes it, but I'm assuming no one else needs it right now...